pandas/tests/test_generic.py

```diff
@@ -1539,6 +1540,15 @@ def test_to_xarray(self):
                            expected,
                            check_index_type=False)
 
+    def test_deepcopy_empty(self):
```

Comment:
Also, I think I might not have made the motivation for this test sufficiently clear.

It's not intended to be a full test of `deepcopy`, for which `_construct` might be useful.  Rather, it's designed specifically to trigger the bug I mentioned in #15370 where the data/index are empty but not the columns.  I don't see how `_construct` would be appropriate for this, but maybe I'm doing it wrong?